core/views.py

```diff
@@ -341,3 +345,82 @@ class SourceDetail(generics.RetrieveUpdateDestroyAPIView):
+
+    def post(self, request, format=None):
+        serializer = ScheduledEventSerializer(data=request.data)
+        if serializer.is_valid():
+            serializer.save()
+            return Response(serializer.data, status=status.HTTP_201_CREATED)
+        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+
+class ScheduledEventRunInstanceList(APIView):
+    """
+    List all ScheduledEventRunInstances, or create a new ScheduledEventRunInstance.
+    """
+    def get(self, request, format=None):
+        scheduled_event_run_instance = ScheduledEventRunInstance.objects.all()
+        serializer = ScheduledEventRunInstanceSerializer(scheduled_event_run_instance, many=True)
+        return Response(serializer.data)
+
+    def post(self, request, format=None):
+        serializer = ScheduledEventRunInstanceSerializer(data=request.data)
+        if serializer.is_valid():
+            serializer.save()
+            return Response(serializer.data, status=status.HTTP_201_CREATED)
+        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+
+class ScheduledEventRunInstanceDetail(APIView):
+    """
+    Get, update or delete ScheduledEventRunInstances using the primary key
+    """
+    def get_object(self, pk):
+        try:
+            return ScheduledEventRunInstance.objects.get(pk=pk)
+        except ScheduledEventRunInstance.DoesNotExist:
+            raise Http404
+
+    def get(self, request, pk, format=None):
+        scheduled_event_run_instance = self.get_object(pk)
+        serializer = ScheduledEventRunInstanceSerializer(scheduled_event_run_instance)
+        return Response(serializer.data)
+
+    def put(self, request, pk, format=None):
+        scheduled_event_run_instance = self.get_object(pk)
+        serializer = ScheduledEventRunInstanceSerializer(scheduled_event_run_instance, data=request.data, partial=True)
+        if serializer.is_valid():
+            serializer.save()
+            return Response(serializer.data)
+        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+    def delete(self, request, pk, format=None):
+        scheduled_event_run_instance = self.get_object(pk)
+        scheduled_event_run_instance.delete()
+        return Response(status=status.HTTP_204_NO_CONTENT)
+
+
+class ScheduledEventRunInstanceHistory(APIView):
+    """
+    Get all ScheduledEventRunInstances related to the provided schedule_event_id
+    """
+    def get_object(self, schedule_event_id):
+        try:
+            return ScheduledEventRunInstance.objects.filter(scheduled_event=schedule_event_id)
+        except ScheduledEventRunInstance.DoesNotExist:
+            raise Http404
+
+    def get(self, request, schedule_event_id, format=None):
+        scheduled_event_run_instance = self.get_object(schedule_event_id)
+        serializer = ScheduledEventRunInstanceSerializer(scheduled_event_run_instance, many=True)
+        return Response(serializer.data)
```

Comment:
Guess this will have to go in favour of one of the others.

Reply:
Updated this class to fetch all run instances for a particular event & add an instance